src/runtime_src/xdp/profile/plugin/device_offload/hal/hal_device_offload_plugin.cpp

```diff
@@ -185,6 +186,13 @@ namespace xdp {
     void* ownedHandle = deviceIdToHandle[deviceId] ;
   
     clearOffloader(deviceId); 
+
+    if (!(db->getStaticInfo()).validXclbin(userHandle)) {
+      xrt_core::message::send(xrt_core::message::severity_level::warning,
+			      "XRT",
+			      "Device profiling is no longer supported on this xclbin.  to enable dvice profiling please rebuild with the current version of the tools.") ;
```

Comment:
I think we should mention the tool version till which we are supporting the xclbins i.e. currently 2019.2. We can add a "double" type const member in Static Info Database and use it while checking valid/supported xclbin and also while reporting this warning. If there is a change in supported version in future, we can just update the const member to the supported version double value.

Reply:
Actually, a method in VPStaticDatabase
double VPStaticDatabase::getOldestSupportedToolVersion { return double(2019.2); }
would be better.

Reply:
Added function to the static database and used it in the construction of the message.